src/SystemView.h

```diff
@@ -168,6 +168,11 @@ class SystemView : public PiGuiView, public DeleteEmitter {
 		Atlas = 1
 	};
 
+	enum class SystemSelectionMode {
+		CurrentSystem = 0,
+		SelectedSystem = 1,
+	};
```

Comment:
This enum should be annotated for `scan_enums.py` to generate `enum_table` entries.

```suggestion
	enum class SystemSelectionMode { // <enum name=SystemSelectionMode scope='SystemView::SystemSelectionMode' public>
		CurrentSystem = 0,
		SelectedSystem = 1,
	};
```

Reply:
Fixed in 62aad68a0b87fcf1a7a9d57a8022668cf50b7a65

----

Generated code should (IMHO) never be checked into source control; instead, the relevant generator should be invoked during configuration and/or build time..(*)  However, requiring this would mean developers would need a version of Python2 installed.. which is currently not the case as long as a developer doesn't add a new enum which should be usable by Lua.

That said, if I'd read the top of the file I guess I would have figured it out too ;) I literally only searched for the other enum and inserted all the required bits for my new one.

(*)  Which, it appears, does happen on system which have a version of Python 2 in the path as `python` - on modern Debian systems, this is not the case.   I fixed this in 5bcbfd6da05003e9b9f5bca91bab517d0b060bd0

Reply:
So there's one specific problem with this approach - using the Ninja generator, every time a rebuild or incremental compile is triggered every file which includes `enum_tables.h` is recompiled as the file has "changed" on disk. Worse yet, it looks like those changes to `enum_tables.h` aren't actually picked up the first time the project is built following a valid change to an annotated enum - it seems CMake (or in this case Ninja) scans dependencies before invoking the `scan_enums.py` script, so it's actually the *second* rebuild that produces a valid program state.

I'd ask that you remove the dependency between `pioneer-core` and the `scan_enums` invocation - instead, leave it defined as a CMake command and we can document it as something that should be run manually in the (very low-frequency) case of modifying an annotated enum value.

Reply:
While I do agree with you in the general case that autogenerated code should likely not be checked into the build system at all, this is a sort of "gray area" where the file typically changes only once every few months, does not have easily-tracked dependencies to determine when it should be re-generated, and touches several distinct areas of the codebase (for example, the script also generates `data/meta/Constants.lua`). As a result it's (historically, in this codebase) "easier" for it to be checked into source control and manually re-generated when actually necessary.

Reply:
I couldn't figure out a way of running it manually and assumed that it just ran at some point during the build process, hence adding the explicit dependency to ensure it ran early in the build.

`ninja enums` gave me an invalid target warning. I'll investigate why.

Good catch that it doesn't build properly the first time around. 

Perhaps instead of having it in the CMake just document that it needs to be invoked? Or run it as part of the build configuration phase?

Reply:
...and trying again now it works as expected, with both `ninja` and `make` invoking the command..  so removing the target dependency, but leaving in the change to properly find Python 2.

Reply:
Thanks! Having it in CMake is mostly so it can be quickly invoked, e.g. via `cmake --build build/ --target enums` or `make -C build enums`, as well as serving as "living documentation" for the correct invocation of the script.